motion/dp_guidance/scripts/dp_guidance.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# Written by Christopher Strom
+# Copyright (c) 2020, Vortex NTNU.
+# All rights reserved.
+
+import rospy
+import actionlib
+
+from move_base_msgs.msg import MoveBaseAction
+from actionlib_msgs.msg import GoalStatus
+
+class DPGuidance:
+    """
+    The dynamic positioning guidance system is very simple.
+    It does not perform any calculations, but simply passes
+    the setpoints directly to the DP controller in the
+    /controller/dp node.
+
+    The dp guidance node is different from the LOS node in the sense
+    that it will not communicate with the controller via a topic.
+    This is done because, as mentioned above, the dp guidance node
+    does not perform any calculations. Therefore, we have no continuous
+    data we need to send, only a discrete goal that will be updated 
+    in the FSM.
+
+    One could argue that there is no need for the dp_guidance node;
+    and you would be right. However, it is introduced as a means of
+    eliminating any crosstalk directly between the fsm and the controller.
+    This will facilitate a cleaner, more modular system in the future.
+    """
+
+    def __init__(self):
+        """
+        Create the ROS node dp and set constants, as well as the action
+        server that the fsm connects to. Connect to the move_base action
+        server in the dp controller. The guidance and controller communicate
+        through this server.
+        """
+
+        rospy.init_node('dp')
+
+        self.timeout = rospy.get_param('~guidance_dp_timeout', 90)
```

Comment:
Not familiar with the code, but is it any difference writing with lower- or uppercase? 'dp' vs. 'DP'

Reply:
Not really, but I've stuck to naming things in lower case